### PR TITLE
Deprecate `NumPyClient`

### PR DESCRIPTION
**Pull request by jafermarq**
This PR reverts #2561.

TODOs:

- [x] Update documentation to reflect `NumPyClient` and `start_numpy_client` is deprecated.
- [x] MXNet-related docs (since we are deprecating mxnet usage in #2724 
- [x] https://github.com/adap/flower/pull/2870
- [x] `.po` files for translations.
- [x] make deprecation message to be rendered correctly in docs.
- [x] Update examples to use `to_client()` -> https://github.com/adap/flower/pull/2718
- [x] Remove usage of `start_numpy_client()` from e2e tests -> https://github.com/adap/flower/pull/2869

### CODE REVIEW

File: doc/locales/fr/LC_MESSAGES/framework-docs.po [hunk 1]
```diff
@@ -14092,7 +14092,7 @@ msgstr ""
 msgid ""
 "That's it for the client. We only have to implement :code:`Client` or "
 ":code:`NumPyClient` and call :code:`fl.client.start_client()` or "
-":code:`fl.client.start_numpy_client()`. The string "
+":code:`fl.client.start_client()`. The string "
```
Comment (danieljanes) [inline doc/locales/fr/LC_MESSAGES/framework-docs.po:14095]: Redundant

File: doc/locales/fr/LC_MESSAGES/framework-docs.po [hunk 2]
```diff
@@ -14101,7 +14101,7 @@ msgid ""
 msgstr ""
 "C'est tout pour le client. Il nous suffit d'implémenter :code:`Client` ou"
 " :code:`NumPyClient` et d'appeler :code:`fl.client.start_client()` ou "
-":code:`fl.client.start_numpy_client()`. La chaîne :code:`\"0.0.0:8080\"` "
+":code:`fl.client.start_client()`. La chaîne :code:`\"0.0.0:8080\"` "
```
Comment (danieljanes) [inline doc/locales/fr/LC_MESSAGES/framework-docs.po:14104]: Redundant

File: doc/locales/fr/LC_MESSAGES/framework-docs.po [hunk 3]
```diff
@@ -14326,7 +14326,7 @@ msgstr ""
 msgid ""
 "That's it for the client. We only have to implement :code:`Client` or "
 ":code:`NumPyClient` and call :code:`fl.client.start_client()` or "
-":code:`fl.client.start_numpy_client()`. The string :code:`\"[::]:8080\"` "
+":code:`fl.client.start_client()`. The string :code:`\"[::]:8080\"` "
```
Comment (danieljanes) [inline doc/locales/fr/LC_MESSAGES/framework-docs.po:14329]: Redundant

File: doc/locales/fr/LC_MESSAGES/framework-docs.po [hunk 4]
```diff
@@ -14335,7 +14335,7 @@ msgid ""
 msgstr ""
 "C'est tout pour le client. Il nous suffit d'implémenter :code:`Client` ou"
 " :code:`NumPyClient` et d'appeler :code:`fl.client.start_client()` ou "
-":code:`fl.client.start_numpy_client()`. La chaîne :code:`\"[: :]:8080\"` "
+":code:`fl.client.start_client()`. La chaîne :code:`\"[: :]:8080\"` "
```
Comment (danieljanes) [inline doc/locales/fr/LC_MESSAGES/framework-docs.po:14338]: Redundant

**Comment by charlesbvll** [comment]
LGTM!

**Review by danieljanes — APPROVED** [review]

**Review by danieljanes — APPROVED** [review]

**Pull request merged**